lib/rmt/scc.rb

```diff
@@ -171,8 +163,8 @@ def create_subscription(item)
   def migration_paths(item)
     product = get_product(item[:id])
     ProductPredecessorAssociation.where(product_id: product.id).destroy_all
-    create_migration_path(product, item[:online_predecessor_ids], :online)
-    create_migration_path(product, item[:offline_predecessor_ids], :offline)
+    create_migration_path(product, item[:online_predecessor_ids], :online) if item[:online_predecessor_ids]
+    create_migration_path(product, item[:offline_predecessor_ids], :offline) if item[:offline_predecessor_ids]
```

Comment:
Wouldn't the serialized JSON have an empty array if there are no predecessors, and thus be always truthy?

Reply:
I believe that is the case. We should check for contents of the array, possibly something like `if !item[:offline_predecessor_ids].empty?`

Reply:
issue with the spec, it was `nil` because wrong spec setup. fixed